src/utils/CometAdapter.cpp

```diff
@@ -0,0 +1,589 @@
+    registerIntOption_("spectrum_batch_size", "<num>", 0, "max. // of spectra to search at a time; 0 to search the entire scan range in one loop", false, true);
+    registerDoubleOption_("mass_offsets", "<offset>", 0, "one or more mass offsets to search (values substracted from deconvoluted precursor mass)", false, true);
+
+    // spectral processing
+    registerIntOption_("minimum_peaks", "<num>", 10, "required minimum number of peaks in spectrum to search (default 10)", false, true);
+    registerIntOption_("minimum_intensity", "<num>", 0, "minimum intensity value to read in", false, true);
+    registerIntOption_("remove_precursor_peak", "<num>", 0, "0=no, 1=yes, 2=all charge reduced precursor peaks (for ETD)", false, true);
+    registerIntOption_("remove_precursor_tolerance", "<num>", 1.5, "+- Da tolerance for precursor removal", false, true);
+    registerStringOption_("clear_mz_range", "[min] [max]", "0.0 0.0", "for iTRAQ/TMT type data; will clear out all peaks in the specified m/z range", false, true);
+
+    //Modifications
+    registerStringList_("fixed_modifications", "<mods>", ListUtils::create<String>(""), "Fixed modifications, specified using UniMod (www.unimod.org) terms, e.g. 'Carbamidomethyl (C)' or 'Oxidation (M)'", false, false);
+    vector<String> all_mods;
+    ModificationsDB::getInstance()->getAllSearchModifications(all_mods);
+    setValidStrings_("fixed_modifications", all_mods);
+    registerStringList_("variable_modifications", "<mods>", ListUtils::create<String>(""), "Variable modifications, specified using UniMod (www.unimod.org) terms, e.g. 'Carbamidomethyl (C)' or 'Oxidation (M)'", false, false);
+    setValidStrings_("variable_modifications", all_mods);
+    registerIntOption_("max_variable_mods_in_peptide", "<num>", 5, "", false, true);
+  }
+
+  vector<ResidueModification> getModifications_(StringList modNames)
+  {
+    vector<ResidueModification> modifications;
+
+    // iterate over modification names and add to vector
+    for (StringList::iterator mod_it = modNames.begin(); mod_it != modNames.end(); ++mod_it)
+    {
+      String modification(*mod_it);
+      modifications.push_back(ModificationsDB::getInstance()->getModification(modification));
+    }
+
+    return modifications;
+  }
+
+  void removeTempDir_(const String& tmp_dir)
+  {
+    if (tmp_dir.empty()) return; // no temp. dir. created
+
+    if (debug_level_ >= 2)
+    {
+      writeDebug_("Keeping temporary files in directory '" + tmp_dir + "'. Set debug level to 1 or lower to remove them.", 2);
+    }
+    else
+    {
+      if (debug_level_ == 1) writeDebug_("Deleting temporary directory '" + tmp_dir + "'. Set debug level to 2 or higher to keep it.", 1);
+      File::removeDirRecursively(tmp_dir);
+    }
+  }
+
+  void createParamFile_(ostream& os)
+  {
+    os << "# comet_version 2016.01 rev. 2\n";               //required as first line in the param file
```

Comment:
> So if you change the version line the param file will work?

Yes, it did for me.

> There is no change in the format of the param file?

Not in the overall format. There may be changes in some parameters, but the tool still works.

> We would have to run comet -p with both versions and check which parameters are added/removed and adjust the param file accordingly.

Here's the diff for the standard settings between "2016.01 rev. 2" and "2017.01 rev. 0beta":
```
$ diff comet.params.2016 comet.params.2017
1c1
< # comet_version 2016.01 rev. 2
---
> # comet_version 2017.01 rev. 0beta
6a7,8
> peff_format = 0                        # 0=no, 1=yes for PSI Extended Fasta Format (PEFF)
> peff_obo =                             # path to UniMod XML for PEFF database
18c20
< isotope_error = 0                      # 0=off, 1=on -1/0/1/2/3 (standard C13 error), 2= -8/-4/0/4/8 (for +4/+8 labeling)
---
> isotope_error = 0                      # 0=off, 1=0/1 (C13 error), 2=0/1/2, 3=0/1/2/3, 4=-8/-4/0/4/8 (for +4/+8 labeling)
69d70
< output_outfiles = 0                    # 0=no, 1=yes  write .out files
84c85
< activation_method = ALL                # activation method; used if activation method set; allowed ALL, CID, ECD, ETD, PQD, HCD, IRMPD
---
> activation_method = ALL                # activation method; used if activation method set; allowed ALL, CID, ECD, ETD, ETD+SA, PQD, HCD, IRMPD
137c138
< add_U_selenocysteine = 0.0000          # added to U - avg. 150.3079, mono. 150.95363
---
> add_U_selenocysteine = 0.0000          # added to U - avg. 150.0379, mono. 150.95363
```

I think those changes are either inconsequential (for the operation of the TOPP tool) or impossible to discover automatically (the change to "isotope_error"). So I think an advanced parameter to set the version, documented with "use at your own risk", should suffice.